lib/mongodb.js

```diff
@@ -615,12 +615,19 @@ MongoDB.prototype.destroyAll = function destroyAll(model, where, callback) {
     where = undefined;
   }
   where = self.buildWhere(model, where);
-  this.collection(model).remove(where || {}, function (err, result) {
-    if (self.debug) {
-      debug('destroyAll.callback', model, where, err, result);
-    }
-    var count = result && result.result && result.result.n || 0;
-    callback && callback(err, count);
+  this.collection(model).remove(where || {}, function(err, results) {
+    if (self.debug)
+      debug('destroyAll.callback', model, where, err, results);
+
+    if (err) return callback(err);
+
+    var info = {
+      count: 0
+    };
+    var affectedCount = results.result.n;
+    if (typeof affectedCount === 'number')
+      info.count = affectedCount;
+    callback && callback(null, info);
```

Comment:
It can be simplified as:
```js
var affectedCount = (results.result && results.result.n) || 0;
callback && callback(null, {count: affectedCount});
```